### ISSUE TITLE
Questions about applying the proposed algorithm to diffusion models

### ISSUE THREAD

**Issue by stonezhng** [VsonicV/es-fine-tuning-paper#40]
Hi, thanks for this awesome work on exploring es algorithm! Just want to ask do you think it is possible to use the same algorithm to finetune diffusion models? Or the inference steps can make the es finetuning more subtle to deal with? Some initial attempts I made show that the target reward is not improving in the first 30 epochs, but due to computational resource constraints the training set only has 4000 image-prompt pairs, so before moving to more engineering work, I want to ask whether it would be possible that it is theoretically infeasible or impractical to use es on diffusion model finetuning?

**Comment by dibbla** [comment]
Hi @stonezhng 

Thanks for your attention to our work! Speaking only for _myself_, I haven't tried diffusion model yet, but I did found some literature relevant during my other works (e.g here is one https://arxiv.org/pdf/2503.14868)

I guess you can try making different perturbation to the models first (as diffusion/flow-matching is repeatedly applying the same model to intermediate states so you might need a smaller perturbation without distroying the image). Feel free to share your finding / question in the discussion section :>

**Comment by stonezhng** [comment]
Thanks so much for pointing out your relevant work! I did observe that more aggressive perturbations lead to generating super noisy images, while smaller perturbations do not change the generated image visually significantly. Still this could be the data size issue as the noisy outputs happen more often during my testing time with just 400 image-prompt pairs

**Comment by dibbla** [comment]
> Thanks so much for pointing out your relevant work! I did observe that more aggressive perturbations lead to generating super noisy images, while smaller perturbations do not change the generated image visually significantly. Still this could be the data size issue as the noisy outputs happen more often during my testing time with just 400 image-prompt pairs

I guess one thing you can consider is the outcome of your model. For example, is the human preference score (or something) is changed after perturbation? Or is the flow-matching loss changed after perturbation? Visual features might not be the most fundamental thing especially consider 1 generation. The goal is to have some difference within the generation that gives you the supervision signal.
